python/instrumentation/openinference-instrumentation-groq/src/openinference/instrumentation/groq/_wrappers.py

```diff
@@ -0,0 +1,185 @@
+    Wrapper for the pipeline processing
+    Captures all calls to the pipeline
+    """
+
+    def __call__(
+        self,
+        wrapped: Callable[..., Any],
+        instance: Any,
+        args: Tuple[Any, ...],
+        kwargs: Mapping[str, Any],
+    ) -> Any:
+        llm_invocation_params = kwargs
+        llm_messages = dict(kwargs).pop("messages", None)
+        if context_api.get_value(context_api._SUPPRESS_INSTRUMENTATION_KEY):
+            return wrapped(*args, **kwargs)
+
+        # Prepare invocation parameters by merging args and kwargs
+        invocation_parameters = {}
+        for arg in args:
+            if arg and isinstance(arg, dict):
+                invocation_parameters.update(arg)
+        invocation_parameters.update(kwargs)
+
+        span_name = "Completions"
+        with self._tracer.start_as_current_span(span_name, attributes={}) as span:
+            span.set_attributes(dict(get_attributes_from_context()))
+
+            attributes = dict(
+                _flatten(
+                    {
+                        SpanAttributes.OPENINFERENCE_SPAN_KIND: LLM,
+                        SpanAttributes.LLM_INPUT_MESSAGES: llm_messages,
+                        SpanAttributes.LLM_INVOCATION_PARAMETERS: safe_json_dumps(
+                            llm_invocation_params
+                        ),
+                        SpanAttributes.LLM_MODEL_NAME: llm_invocation_params.get("model"),
+                        SpanAttributes.INPUT_MIME_TYPE: OpenInferenceMimeTypeValues.JSON,
+                    }
+                )
+            )
+            try:
+                response = wrapped(*args, **kwargs)
+            except Exception as exception:
+                span.set_status(trace_api.Status(trace_api.StatusCode.ERROR, str(exception)))
+                span.record_exception(exception)
+                raise
+            attributes.update(
+                _flatten(
+                    {
+                        SpanAttributes.OUTPUT_VALUE: response.choices[0].message.content,
+                        SpanAttributes.OUTPUT_MIME_TYPE: OpenInferenceMimeTypeValues.TEXT,
+                        SpanAttributes.LLM_TOKEN_COUNT_COMPLETION: response.usage.completion_tokens,
+                        SpanAttributes.LLM_TOKEN_COUNT_PROMPT: response.usage.prompt_tokens,
+                        SpanAttributes.LLM_TOKEN_COUNT_TOTAL: response.usage.total_tokens,
+                    }
+                )
+            )
+
+            span.set_attributes(attributes)
+            span.set_status(trace_api.StatusCode.OK)
+
+        return response
+
+
+class _AsyncCompletionsWrapper(_WithTracer):
+    """
+    Wrapper for the pipeline processing
+    Captures all calls to the pipeline
+    """
+
+    async def __call__(
+        self,
+        wrapped: Callable[..., Any],
+        instance: Any,
+        args: Tuple[Any, ...],
+        kwargs: Mapping[str, Any],
+    ) -> Any:
+        comp_type = instance.__class__.__name__
```

Comment:
we should check `context_api.get_value(context_api._SUPPRESS_INSTRUMENTATION_KEY)` here too right?